lib/api/src/rc_switch_user.cpp

```diff
@@ -15,11 +15,31 @@ auto rc_switch_user(RcComm* _comm, const char* _username, const char* _zone) ->
         return SYS_INVALID_INPUT_PARAM;
     }
 
+    // Accepting empty strings for either parameter is not allowed.
+    // This is especially true for the "_zone" parameter because the implementation of this
+    // function copies the string represented by "_zone" into the RcComm. Copying an empty
+    // string into the RcComm's rodsZone member variables makes the RcComm invalid.
+    if (std::strlen(_username) == 0 || std::strlen(_zone) == 0) {
+        return SYS_INVALID_INPUT_PARAM;
+    }
+
     auto& client = _comm->clientUser;
 
-    if (std::strcmp(_username, client.userName) == 0 && std::strcmp(_zone, client.rodsZone) == 0) {
+    // clang-format off
+    // NOLINTBEGIN(cppcoreguidelines-pro-bounds-array-to-pointer-decay)
+    //
+    // Return immediately if the RcComm object already represents the target user.
+    //
+    // This is allowed because the RcComm only cares about the username and zone. If the RcComm
+    // considered the user's type or authentication level (i.e. LOCAL_PRIV_USER_AUTH), then this
+    // optimization would not be possible.
+    if (std::strncmp(_username, client.userName, sizeof(UserInfo::userName)) == 0 &&
```

Comment:
the use of `strncmp` is to be more specific/safe since it only checks `n` characters?

is there any other reason here?

Reply:
Correct.

The struct, `UserInfo`, defines the exact size of a username. Therefore, we check up to that many characters.

Thinking about this more, we should probably check `sizeof(UserInfo::userName) - 1` bytes. That takes the null byte into account.

Reply:
definitely poke at it with the edge cases and document any magical `- 1`s in the code please.

Reply:
Assuming the value stored in `UserInfo::userName` is expected to be a null-terminated string. My guess is yes.

Reply:
done